mathics/doc/common_doc.py

```diff
@@ -138,6 +138,31 @@
 test_result_map = {}
 
 
+# TODO: Use this when checking requires for modules.
+requires_lib_cache = {}
+
+
+def check_requires_list(requires: list) -> bool:
+    """
+    Check if module names in ``requires`` can be imported and return True if they can or False if not.
+
+    This state value is also recorded in dictionary `requires_lib_cache` keyed by module name and is used to determine whether to skip trying to get information from the module."""
+    global requires_lib_cache
```

Comment:
This is what I had in mind. Having or not this line does not seems to change anything, just to state that `require_lib_cache` is a variable at the level of the module.

Reply:
Please don't add a global. It is frowned upon generally.  Some would argue that this should be passed as a parameter, and that is the typical way you convey this - in the interface. 


I know this kind of thing, a creative use of  `global` seems innovated. I used to (and still sometimes do) add `pass` statements where in most programming languages an end closer would appear. However it puts off most programmers so I generally don't do it except in code where I am pretty much the only one who will ever use this. 

There is a place for creativity, but here and this way sadly isn't the place for that.

Reply:
Regarding this, probably the cache for the previous results is not needed: Python have also a cache inside `importlib`, so the global variable does not provide too much advantage.